core/src/main/java/org/opensearch/sql/expression/function/udf/ToNumberFunction.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.expression.function.udf;
+
+import java.math.BigInteger;
+import java.util.List;
+import org.apache.calcite.adapter.enumerable.NotNullImplementor;
+import org.apache.calcite.adapter.enumerable.NullPolicy;
+import org.apache.calcite.adapter.enumerable.RexToLixTranslator;
+import org.apache.calcite.linq4j.function.Strict;
+import org.apache.calcite.linq4j.tree.Expression;
+import org.apache.calcite.linq4j.tree.Expressions;
+import org.apache.calcite.rex.RexCall;
+import org.apache.calcite.sql.type.SqlReturnTypeInference;
+import org.apache.calcite.sql.type.SqlTypeName;
+import org.opensearch.sql.calcite.utils.PPLOperandTypes;
+import org.opensearch.sql.expression.function.ImplementorUDF;
+import org.opensearch.sql.expression.function.UDFOperandMetadata;
+
+/**
+ * A custom implementation of number/boolean to string .
+ *
+ * <p>This operator is necessary because tostring has following requirements "binary" Converts a
+ * number to a binary value. "hex" Converts the number to a hexadecimal value. "commas" Formats the
+ * number with commas. If the number includes a decimal, the function rounds the number to nearest
+ * two decimal places. "duration" Converts the value in seconds to the readable time format
+ * HH:MM:SS. if not format parameter provided, then consider value as boolean
+ */
+public class ToNumberFunction extends ImplementorUDF {
+  public ToNumberFunction() {
+    super(
+        new org.opensearch.sql.expression.function.udf.ToNumberFunction.ToNumberImplementor(),
+        NullPolicy.ANY);
+  }
+
+  @Override
+  public SqlReturnTypeInference getReturnTypeInference() {
+    return (opBinding) -> {
+      // Try to determine if the result will be Long or Double based on the input
+      int base = 10;
+      try {
+        base =
+            opBinding.getOperandCount() > 1
+                ? opBinding.getOperandLiteralValue(1, Integer.class)
+                : 10;
+      } catch (NumberFormatException e) {
+        // If parsing fails, default to  base 10
+      }
```

Comment:
We've provided type checker in `getOperandMetadata`. so the second argument must be numeric right?